doc/source/whatsnew/v2.1.0.rst

```diff
@@ -232,6 +232,7 @@ Numeric
 ^^^^^^^
 - Bug in :meth:`Series.corr` and :meth:`Series.cov` raising ``AttributeError`` for masked dtypes (:issue:`51422`)
 - Bug in :meth:`DataFrame.corrwith` raising ``NotImplementedError`` for pyarrow-backed dtypes (:issue:`52314`)
+- Bug in :func:`_read`, set dtype_backend to "pyarrow" if engine is "pyarrow" (:issue:`52505`)
```

Comment:
Could you rephrase this describing the issue with `read_csv` from a user perspective?